config/config.go

```diff
@@ -414,12 +418,23 @@ func parseHcl(parser *hclparse.Parser, hcl string, filename string) (file *hcl.F
 		}
 	}()
 
-	file, parseDiagnostics := parser.ParseHCL([]byte(hcl), filename)
-	if parseDiagnostics != nil && parseDiagnostics.HasErrors() {
-		return nil, parseDiagnostics
+	if filepath.Ext(filename) == ".hcl" {
+		file, parseDiagnostics := parser.ParseHCL([]byte(hcl), filename)
+		if parseDiagnostics != nil && parseDiagnostics.HasErrors() {
+			return nil, parseDiagnostics
+		}
+	
+		return file, nil
+	}else if filepath.Ext(filename) == ".json" {
+		file, parseDiagnostics := parser.ParseJSON([]byte(hcl), filename)
+		if parseDiagnostics != nil && parseDiagnostics.HasErrors() {
+			return nil, parseDiagnostics
+		}
+	
+		return file, nil
```

Comment:
You are getting that because it won't fit the type. Without it, you have a code path where you don't return anything (when `filename` doesn't end with `hcl` or `json`). Golang requires you to exhaust all the options in the static type checking so you need that last `else` block.

Note that if you follow the suggestion above, where you only have the `json` condition, that last `else` block goes away.